### PR TITLE
Add snap configuration file

### PR DESCRIPTION
**Pull request by nicolasbock**
Closes: #1992

Signed-off-by: Nicolas Bock <nicolas.bock@canonical.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [ ] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [ ] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

File: snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: sosreport
```
Comment (TurboTurtle) [inline snapcraft.yaml:1]: For new packaging, I think using the project name of `sos` is a better approach, as the project is now more than the report functionality.

File: .travis.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
       dist: bionic
       language: python
       python: "3.6"
-      install: pip install -r requirements.txt; python3 setup.py install;
+      install: pip install -r test-requirements.txt; python3 setup.py install;
```
Comment (TurboTurtle) [inline .travis.yml:34]: I don't see why we'd need to split the requirements into 2 files. Not least of which, this stops pulling in pexpect which once we have testing for `collect` figured out, we'd need to add it to the new file as well. I don't see a benefit in splitting out the requirements for a single different module to install.

**Review by TurboTurtle — CHANGES_REQUESTED** [review]
I realize this is still a draft, but a few comments below.

Additionally, please include an informative commit message beyond an issue reference, for historical tracking when we look back at this in the future.

File: snapcraft.yaml (hunk 1 above)
Comment (nicolasbock) [reply snapcraft.yaml:1]: :+1:

File: .travis.yml (hunk 2 above)
Comment (nicolasbock) [reply .travis.yml:34]: I am still experimenting here. The main reason I split the requirements though is that packages such as `pycodestyle` are not necessary for `sos` and are only needed for testing (unless I am missing something).

**Comment by nicolasbock** [comment]
I am getting an error when building the snap:
```
/root/parts/sosreport/install/usr/bin/python3 setup.py --no-user-cfg install --single-version-externally-managed --user --record install.txt
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: option --single-version-externally-managed not recognized
```

It's not clear to me what this error message means at this point...

**Comment by nicolasbock** [comment]
@TurboTurtle Eventually, once this here is working properly, the `sos` project needs to register the `sos` snap in the snap store [1].Please note also that the snap requires `classic` confinement which means less sandboxing basically so snapped `sos` can access the various system files it needs to access. However, `classic` confinement is discouraged because of the potential security implications but can be requested for apps that actually need it. I have done so [2].

[1] https://snapcraft.io/register-snap
[2] https://forum.snapcraft.io/t/request-for-classic-confinement-for-sos/19912

**Comment by TurboTurtle** [comment]
Wanted to cycle back around on this, is this still desired for inclusion in 4.1? What's outstanding for this change?

**Comment by nicolasbock** [comment]
Closing because of #2676

**Pull request closed without merging**